dsls/src/main/scala/ch/epfl/directembedding/test/TestBase.scala

```diff
@@ -12,13 +12,16 @@ case class JustArgs(x: Exp[Int]) extends Exp[Int]
 case class ArgsAndTArgs[T, U](t: Exp[T], u: Exp[U]) extends Exp[(T, U)]
 case class Const[T](x: T) extends Exp[T]
 
-case class Size[T](self: Exp[TArgClassExampleCase[T]]) extends Exp[Int]
-case class Take[T](self: Exp[TArgClassExampleCase[T]], n: Exp[Int]) extends Exp[TArgClassExample[T]]
-case class X[T](self: Exp[TArgClassExample[T]]) extends Exp[Int]
-case class Y[T](self: Exp[TArgClassExample[T]]) extends Exp[Int]
-case class TArgsZ[T, U](self: Exp[TArgClassExample[T]]) extends Exp[U]
-case class AppCurry[T](self: Exp[TArgClassExample[T]], p1: Exp[T], p2: Exp[T], p3: Exp[T]*) extends Exp[T]
-case class AppManyArgs[T](self: Exp[TArgClassExample[T]], p1: Exp[T]*) extends Exp[T]
+case class Size[T](self: Exp[_]) extends Exp[Int]
+case class Take[T](self: Exp[_], n: Exp[Int]) extends Exp[TArgClassExample[T]]
```

Comment:
To allow nested `Take` objects, see [this test](https://github.com/directembedding/directembedding/pull/23/files#diff-5079bd066de90e7709252650057024efR101)